datadog/dogstatsd/base.py

```diff
@@ -978,9 +1036,16 @@ def _report(self, metric, metric_type, value, tags, sample_rate):
         if sample_rate != 1 and random() > sample_rate:
             return
 
+        # timestamps (protocol v1.3) only allowed on gauges and counts
+        allows_timestamp = metric_type == "g" or metric_type == "c"
+        if not allows_timestamp or timestamp < 0:
+            timestamp = 0
```

Comment:
Do you think that would be worth a warning? It's a developer error at this point (somebody is calling `_report` with a wrong `metric_type`)

Reply:
Yes worth a test (as said in the other comment) but maybe we don't want to flood stdout with logs?

Reply:
I added a test but not sure about adding a warning. In other SDKs we don't log a warning if invalid timestamp used. Here if they are trying to send a metric that doesn't support timestamps then they're accessing methods they shouldn't be in the first place.